btsieve/src/bitcoin/blockchain_info_hex_bitcoin_http_blocksource.rs

```diff
@@ -0,0 +1,169 @@
+use crate::blocksource::{self, BlockSource};
+use bitcoin_support::{deserialize, MinedBlock, Network};
+use futures::{Future, Stream};
+use reqwest::r#async::Client;
+use serde::Deserialize;
+use std::time::Duration;
+use tokio::timer::Interval;
+
+#[derive(Deserialize)]
+struct BlockchainInfoLatestBlock {
+    hash: String,
+    height: u32,
+}
+
+#[derive(Debug)]
+pub enum Error {
+    UnsupportedNetwork(String),
+    Reqwest(reqwest::Error),
+    Hex(hex::FromHexError),
+    BlockDeserialization(String),
+}
+
+#[derive(Clone)]
+pub struct BlockchainInfoHexHttpBlockSource {
+    network: Network,
+    client: Client,
+}
+
+impl BlockchainInfoHexHttpBlockSource {
+    pub fn new(network: Network) -> Result<Self, Error> {
+        // Currently configured for Mainnet only because no support for hex-encoded
+
+        match network {
+            Network::Mainnet => Ok(Self {
+                network,
+                client: Client::new(),
+            }),
+            _ => {
+                log::error!(
+                    "Network {} not supported for bitcoin http blocksource",
+                    network
+                );
+                Err(Error::UnsupportedNetwork(format!(
+                    "Network {} currently not supported for bitcoin http plocksource",
+                    network
+                )))
+            }
+        }
+    }
+
+    pub fn latest_block(&self) -> impl Future<Item = MinedBlock, Error = Error> + Send + 'static {
+        let cloned_self = self.clone();
+
+        self.latest_block_without_tx()
+            .and_then(move |latest_block| {
+                cloned_self.raw_hex_block(latest_block.hash, latest_block.height)
+            })
+    }
+
+    fn base_url(&self) -> String {
+        match self.network {
+            Network::Mainnet => "https://blockchain.info".to_string(),
+            _ => panic!(
+                "Network {} not supported for bitcoin.info blocksource",
+                self.network
+            ),
+        }
+    }
+
+    fn latest_block_without_tx(
+        &self,
+    ) -> impl Future<Item = BlockchainInfoLatestBlock, Error = Error> + Send + 'static {
+        // https://blockchain.info/q/latesthash only works for mainnet, there is no testnet endpoint
+        // we fall-back to [testnet.]blockchain.info/latestblock to retrieve the latest
```

Comment:
So there is a testnet endpoint? I don't understand the limitation.

Reply:
There is no testnetnet endpoint returning the full hash of a block unfortunately